From ae526a4276abeccdd0b0c738dea903c5575a52b6 Mon Sep 17 00:00:00 2001
From: Mamoru TASAKA <mtasaka@fedoraproject.org>
Date: Sun, 23 Oct 2022 02:41:03 +0900
Subject: [PATCH] demo-Gtk.c/populate_prefs_page: use correct pointer for
 pref_changed_cb

With xscreensaver 6.05. xscreensaver-settings causes segfault when
trying to choose another theme for auth window. This is because
in populate_prefs_page() wrong pointer is passed to pref_changed_cb(),
then XSCREENSAVER_WINDOW cast complains.

The first line in populate_prefs_page() actually writes what pointer
should be used - use the same pointer here.
---
 driver/demo-Gtk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/driver/demo-Gtk.c b/driver/demo-Gtk.c
index 404fc88..aa85923 100644
--- a/driver/demo-Gtk.c
+++ b/driver/demo-Gtk.c
@@ -2518,7 +2518,7 @@ populate_prefs_page (state *s)
         if (! signal_connected_p)
           {
             g_signal_connect (G_OBJECT (cbox), "changed",
-                              G_CALLBACK (pref_changed_cb), (gpointer) s);
+                              G_CALLBACK (pref_changed_cb), (gpointer) s->window);
             signal_connected_p = TRUE;
           }
       }
-- 
2.38.1

